ui/component/publishFile/view.jsx

```diff
@@ -237,53 +309,84 @@ function PublishFile(props: Props) {
     };
     // Strip off extention and replace invalid characters
     let fileName = name || file.name.substr(0, file.name.lastIndexOf('.')) || file.name;
-    let INVALID_URI_CHARS = new RegExp(regexInvalidURI, 'gu');
-    let parsedFileName = fileName.replace(INVALID_URI_CHARS, '-');
+
     if (!isStillEditing) {
-      publishFormParams.name = parsedFileName;
+      publishFormParams.name = parseName(fileName);
+    }
+
+    // Prevent name autopopulation from title
+    if (autoPopulateName) {
+      setAutoPopulateName(false);
     }
+
     // File path is not supported on web for security reasons so we use the name instead.
     setCurrentFile(file.path || file.name);
     updatePublishForm(publishFormParams);
   }
 
-  let title;
+  let cardTitle;
   if (publishing) {
-    title = (
+    cardTitle = (
       <span>
         {__('Uploading')}
         <Spinner type={'small'} />
       </span>
     );
   } else {
-    title = isStillEditing ? __('Edit') : __('Upload');
+    cardTitle = isStillEditing ? __('Edit') : __('Upload');
   }
 
+  const isPublishFile = mode === PUBLISH_MODES.FILE;
+  const isPublishPost = mode === PUBLISH_MODES.POST;
+
   return (
     <Card
       icon={ICONS.PUBLISH}
       disabled={disabled || balance === 0}
       title={
         <React.Fragment>
```

Comment:
Can you try adding the "file"/"post" toggle here? So it matches the style on the discover page with the toggle _inside_ of the card. It might look weird above the title but I would still like to try it.

Reply:
where exactly ? if I add it before the title, the icon don't look good in that position:
![image](https://user-images.githubusercontent.com/14793624/88963227-2909d580-d26d-11ea-8376-6a78ee98672a.png)

Reply:
IMO it don't belong in that card  because it also affects other cards, like the description card.

Reply:
what if we put it above all the cards, before upload entirely?

then it's like two tabs: [Upload / Post]

Reply:
that's the current design ^^

Reply:
This looks better if you really want it inside the card 😛 
![image](https://user-images.githubusercontent.com/14793624/88964276-caddf200-d26e-11ea-8c50-f9d355b50e20.png)

Reply:
Sorry I didn't respond. I've been away from my computer for a few days. I think your original design is better. Thanks for showing what this would look like though.